src/layers/grid-layer/grid-layer.js

```diff
@@ -55,89 +54,28 @@ export default class GridLayer extends AggregationLayer {
     return GridLayerIcon;
   }
 
-  formatLayerData(datasets, oldLayerData, opt = {}) {
-    const formattedData = super.formatLayerData(
-      datasets,
-      oldLayerData,
-      opt
-    );
+  renderLayer(opts) {
+    const {
+      data,
+      objectHovered,
+      mapState
+    } = opts;
 
-    const {getPosition, data} = formattedData;
-
-    // TODO: fix this in deck.gl layer
-    const cleaned = data.filter(d => {
-      const pos = getPosition(d);
-      return pos.every(Number.isFinite);
-    });
-
-    // All data processing is done in deck.gl layer
-    return {
-      ...formattedData,
-      data: cleaned
-    };
-  }
-
-  renderLayer({
-    data,
-    idx,
-    objectHovered,
-    mapState,
-    interaction,
-    layerCallbacks,
-    layerInteraction
-  }) {
     const zoomFactor = this.getZoomFactor(mapState);
-    const eleZoomFactor = this.getElevationZoomFactor(mapState);
     const {visConfig} = this.config;
     const cellSize = visConfig.worldUnitSize * 1000;
 
     return [
-      new DeckGLGridLayer({
+      new EnhancedGridLayer({
+        ...this.getDefaultAggregationLayerProp(opts),
         ...data,
-        ...layerInteraction,
-        id: this.id,
-        idx,
-        cellSize,
-        coverage: visConfig.coverage,
-        // highlight
-        autoHighlight: visConfig.enable3d,
-        highlightColor: HIGHLIGH_COLOR_3D,
-
-        // color
-        colorRange: this.getColorRange(visConfig.colorRange),
-        colorScale: this.config.colorScale,
-        sizeScale: this.config.sizeScale,
-        opacity: visConfig.opacity,
-        upperPercentile: visConfig.percentile[1],
-        lowerPercentile: visConfig.percentile[0],
-
-        // elevation
-        extruded: visConfig.enable3d,
-        elevationScale: visConfig.elevationScale * eleZoomFactor,
-        elevationRange: visConfig.sizeRange,
-        elevationLowerPercentile: visConfig.elevationPercentile[0],
-        elevationUpperPercentile: visConfig.elevationPercentile[1],
-        // parameters
-        parameters: {depthTest: Boolean(visConfig.enable3d || mapState.dragRotate)},
-
-        // render
-        pickable: true,
-
-        // callbacks
-        onSetColorDomain: layerCallbacks.onSetLayerDomain,
-
-        _subLayerProps: {
-          CPU: {
-            type: EnhancedCPUGridLayer
-          }
-        }
+        cellSize
       }),
 
       // render an outline of each cell if not extruded
       ...(this.isLayerHovered(objectHovered) && !visConfig.enable3d
         ? [
             new GeoJsonLayer({
-              ...layerInteraction,
               id: `${this.id}-hovered`,
```

Comment:
should we use ...this.getDefaultHoverLayerProps()?

Reply:
👍